src/app/components/locationtree/tree.component.ts

```diff
@@ -65,9 +65,7 @@ export class TreeNodeComponent {
         .subscribe(
           result => {
             this.childLocations.push(result);
-            if (i === children.length - 1) {
-              this.childLocations.sort(this.compareChildren);
-            }
+            this.childLocations.sort(this.compareChildren);
```

Comment:
Should probably note (maybe in code) the runtime complexity/performance hit that this fix incurs, for future optimization.